addon/serializers/json-api-serializer.js

```diff
@@ -136,6 +136,12 @@ const JSONAPISerializer = JSONSerializer.extend({
     @private
   */
   _normalizeResourceHelper: function(resourceHash) {
+    if (Ember.isNone(resourceHash.type)) {
+      Ember.assert(this.warnMessageForUndefinedType(), false, {
```

Comment:
Do you mind changing the `false` here to `!Ember.isNone(resourceHash.type)` and remove the if statement that way when the production build strips the `assert` there won't be an empty if statement.